### PR TITLE
Add ViteJS as the frontend dev server and build system

### PR DESCRIPTION
**Pull request by dherault**
## Summary

This PR adds ViteJS as the dev server and build system for the frontend.
It should be 100% compatible with our build and deploy infra.

@michaeljguarino can you test the cloud shell by checking out this branch because I couldn't do it.

Results: 21s build with Webpack --> Less than a second with Vite

NOTE: This PR is based on previous DS work because I needed to take into account every open frontend PRs.

Resolves ENG-948
## Checklist

- [ ] If required, I have updated the Plural documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] I have added a meaningful title and summary to convey the impact of this PR to a user.
- [x] I have added relevant labels to this PR to help with categorization for release notes.

### CODE REVIEW

**Comment by dherault** [comment]
@floreks Could you check what is wrong with the Unit and E2E tests?

File: www/package.json [hunk 1]
```diff
@@ -1,12 +1,15 @@
 {
   "name": "www",
-  "version": "0.1.0",
   "private": true,
+  "version": "0.1.0",
+  "type": "module",
   "scripts": {
+    "dev": "vite",
```
Comment (maciaszczykm) [inline www/package.json:7]: There is `start` already, do we need both?

File: www/package.json [hunk 2]
```diff
@@ -39,28 +42,32 @@
...
+    "honorable-theme-default": "0.66.0",
     "js-yaml": "4.1.0",
     "lodash": "4.17.21",
     "memoize-one": "6.0.0",
     "moment": "2.29.4",
     "phoenix": "1.6.15",
     "pluralsh-absinthe-socket-apollo-link": "0.2.0",
-    "pluralsh-design-system": "1.241.0",
+    "pluralsh-design-system": "1.288.0",
+    "process": "^0.11.10",
```
Comment (maciaszczykm) [inline www/package.json:70]: Can you change to exact versions here and for other packages?

**Comment by maciaszczykm** [comment]
Units might be failing because of changes in Babel configuration. Yesterday @floreks mentioned [Vitest](https://vitest.dev/). We might use it as it should be easier to integrate with current setup than Jest.

**Comment by floreks** [comment]
Ye, I will look into switching to vitest.

**Comment by floreks** [comment]
Switched to using `vitest` instead of `jest`. Unit tests should now be passing.

**Comment by floreks** [comment]
Apparently, `vite` dev server did not want to play well with the `wait-on` package. I have added a workaround to make it work. E2E should now pass.

**Comment by dherault** [comment]
Thank you @floreks !!!

File: www/package.json (hunk 1 above)
Comment (dherault) [reply www/package.json]: Done

File: www/package.json (hunk 2 above)
Comment (dherault) [reply www/package.json]: Done by @floreks

**Review by maciaszczykm — APPROVED** [review]
Great! I think lockfile needs to be regenerated to fix CI.

**Pull request merged**